### ISSUE TITLE
Loses "this" context in getter function with "compress.inline"

### ISSUE THREAD

**Issue by zalishchuk** [swc-project/swc#5914]
### Describe the bug

Seems like the minifier loses the context when trying to minify the getter function which refers to upper `this` context through `that` variable, and for some reason it shrinks the getter body to `return this.config` which in result causes an `Uncaught RangeError: Maximum call stack size exceeded` error in this case. The option that causes this error is probably `compress.inline`. Also, if you remove the getter keyword and execute `instance.options.config` as a function - everything will be fine. Terser in this case works fine even with `compress.inline` option enabled.

### Input code

```typescript
class Test {
  constructor(config) {
    const that = this;
    this.config = config;
    this.options = {
      get config() {
        return that.config;
      },
    };
  }
}

const instance = new Test(42);
console.log(instance.options.config); // should log 42
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "tsx": false
    },
    "target": "es5",
    "loose": false,
    "minify": {
      "compress": {
        "arguments": false,
        "arrows": true,
        "booleans": true,
        "booleans_as_integers": false,
        "collapse_vars": true,
        "comparisons": true,
        "computed_props": true,
        "conditionals": true,
        "dead_code": true,
        "directives": true,
        "drop_console": false,
        "drop_debugger": true,
        "evaluate": true,
        "expression": false,
        "hoist_funs": false,
        "hoist_props": true,
        "hoist_vars": false,
        "if_return": true,
        "join_vars": true,
        "keep_classnames": false,
        "keep_fargs": true,
        "keep_fnames": false,
        "keep_infinity": false,
        "loops": true,
        "negate_iife": true,
        "properties": true,
        "reduce_funcs": false,
        "reduce_vars": false,
        "side_effects": true,
        "switches": true,
        "typeofs": true,
        "unsafe": false,
        "unsafe_arrows": false,
        "unsafe_comps": false,
        "unsafe_Function": false,
        "unsafe_math": false,
        "unsafe_symbols": false,
        "unsafe_methods": false,
        "unsafe_proto": false,
        "unsafe_regexp": false,
        "unsafe_undefined": false,
        "unused": true,
        "const_to_let": true,
        "pristine_globals": true
      },
      "mangle": {
        "toplevel": false,
        "keep_classnames": false,
        "keep_fnames": false,
        "keep_private_props": false,
        "ie8": false,
        "safari10": false
      }
    }
  },
  "module": {
    "type": "es6"
  },
  "minify": true,
  "isModule": true,
  "env": {
    "targets": ""
  }
}
```

### Playground link

https://play.swc.rs/?version=1.3.2&code=H4sIAAAAAAAAA02OQQrDIBRE955ilgrFQMgu9Ba9gFibCKJFf%2BiiePf%2BaFKy0XFmfIwNphQ8XCF8BWBTLJQ3SylL1i%2B%2FqOYfCWg1hDtfvszN3pXuTfa7uCTpTZ4%2FctQpOCpS%2FQ0gO9pybGx9JQD11kTd31VUIfoKz4eJ1jE2uk9bL6dRzS1OwemQFnmWzg0HWs0YBpQ1beEJ7mEaf5Vzr8AEAQAA&config=H4sIAAAAAAAAA3VUy5LcIAz8F857SA5JpfYDcss3UIwRHjYYuZDwjmvK%2Fx7hB4xnNjeraQlJ3eauPqhT73c1mkSQyhfNkc1NvSueR6Au%2BZHVm2ISyJlAsEhgUg8sFKAfchYQCfbTNzX46N1cKnU4jAmIyrdk5AEiU%2BWZlPBTQk5ZogtiABNfYm1I%2B8jQQ2qpHYZgRgI9mVQzym0mecJ4gjKD1WPC8QGN1rPHKMUOzIKxukMLFfAJOvYTNIrUEEokaax2soIWLrnvy%2FY2JkwmZMO1FtzWNciNNe%2BKnli7HOkJOnW6QduQO8s7nYBzigfnA3087eEvgPQZDFE0A7TMFXeiw5npvmD56EREnisqCreuIvQynPbe1QlL15DYt20lsLmDMmHXiu%2FgaSDyFjQ4J9uuyfTpubu2YsWJ6GooWzOuabCF%2BvDTGS0WeAF%2FS1f8KMeOD4avzxjNwwXDS4kB%2BIr2BZZFMD6DSdx7G5%2FRHC3ImsE%2BHGQqYbWpiM%2BoQ%2FnVjj2LISRH9wEv1b7yRw4m9sWWd8U4BpggnBX9rx%2B%2BwOSSqQi8W%2FEwHvxqmhnxkf%2F%2B7XgRltIB2rx3IGqtb8NPtbTnYBvA05%2Bdt%2F8acVpT1gdFLlNqWf4B%2FNGgiJQEAAA%3D

### Expected behavior

```js
var n = new (function n(o) {
  "use strict";
  !(function (n, o) {
    if (!(n instanceof o))
      throw new TypeError("Cannot call a class as a function");
  })(this, n);
  var i = this;
  (this.config = o),
    (this.options = {
      get config() {
        return i.config;
      },
    });
})(42);
console.log(n.options.config);
```

### Actual behavior

```js
var n = new (function n(o) {
  "use strict";
  !(function (n, o) {
    if (!(n instanceof o))
      throw new TypeError("Cannot call a class as a function");
  })(this, n),
    (this.config = o),
    (this.options = {
      get config() {
        return this.config;
      },
    });
})(42);
console.log(n.options.config);
```

### Version

1.3.2

**Comment by kdy1** [comment]
Your input does not print 42

**Comment by zalishchuk** [comment]
@kdy1 sorry, missed the `get` keyword in the input, fixed it

**Comment by zalishchuk** [comment]
@kdy1 Still a problem in v1.3.3, [playground link](https://play.swc.rs/?version=1.3.3&code=H4sIAAAAAAAAA02PQQrDIBRE955ilgrFQMhOeoteIFibCKJFf%2BiiePf%2BaFKy0XFm%2FuNrw1wKHq4QvgKwKRbKm6WUJeuXX1TzjwS0zoQ7X76YZu9K9yb7XVyS9CbPgxx1CrA4OmpS%2FU0gO9pybHx9pQD11kTd31VUIfomno85Wsfo6D7tB3IalWlxCk6HtMizdO5xoJXBMKCsaQtPcA%2FT%2BAN8ZTP9CAEAAA%3D%3D&config=H4sIAAAAAAAAA3VUy5LcIAz8F857SA5JpfYDcss3UIwRHjYYuZDwjmvK%2Fx7hB4xnNjeraQlJ3eauPqhT73c1mkSQyhfNkc1NvSueR6Au%2BZHVm2ISyJlAsEhgUg8sFKAfchYQCfbTNzX46N1cKnU4jAmIyrdk5AEiU%2BWZlPBTQk5ZogtiABNfYm1I%2B8jQQ2qpHYZgRgI9mVQzym0mecJ4gjKD1WPC8QGN1rPHKMUOzIKxukMLFfAJOvYTNIrUEEokaax2soIWLrnvy%2FY2JkwmZMO1FtzWNciNNe%2BKnli7HOkJOnW6QduQO8s7nYBzigfnA3087eEvgPQZDFE0A7TMFXeiw5npvmD56EREnisqCreuIvQynPbe1QlL15DYt20lsLmDMmHXiu%2FgaSDyFjQ4J9uuyfTpubu2YsWJ6GooWzOuabCF%2BvDTGS0WeAF%2FS1f8KMeOD4avzxjNwwXDS4kB%2BIr2BZZFMD6DSdx7G5%2FRHC3ImsE%2BHGQqYbWpiM%2BoQ%2FnVjj2LISRH9wEv1b7yRw4m9sWWd8U4BpggnBX9rx%2B%2BwOSSqQi8W%2FEwHvxqmhnxkf%2F%2B7XgRltIB2rx3IGqtb8NPtbTnYBvA05%2Bdt%2F8acVpT1gdFLlNqWf4B%2FNGgiJQEAAA%3D). Did you apply any fixes for this problem in v1.3.3 version?

**Comment by kdy1** [comment]
No issue was wrong. Please file a new issue with appropriate innput
